portal/src/graphql/adminapi/AddRolesScreen.tsx

```diff
@@ -111,8 +106,15 @@ const AddRolesScreen: React.VFC = function AddRolesScreen() {
     };
   }, [setFormState]);
 
+  useEffect(() => {
+    if (form.submissionResult != null) {
+      const roleID = form.submissionResult;
+      navigate(`../${encodeURIComponent(roleID)}/details`, { replace: true });
+    }
+  }, [form.submissionResult, navigate]);
+
```

Comment:
It is because we need submissionResult, which is not available in afterSave.

In detail:
If we use `submissionResult` in `afterSave`, `afterSave` must be regenerated to use the latest value of `submissionResult`. However, the latest instance of `afterSave` will not be used after save, because the previous instance of `afterSave` was captured..

Reply:
> However, the latest instance of afterSave will not be used after save, because the previous instance of afterSave was captured.

Sounds like a bug to me :sosad:

Reply:
Yes, but it is a quite common problem of react hooks, known as the stale closure.
Fixing it requires some hacky code such as using a ref to store the latest `afterSave`, which I think using useEffect is better.